cerberus/validator.py

```diff
@@ -1038,8 +1058,17 @@ def _validate_keyschema(self, schema, field, value):
     def _validate_readonly(self, readonly, field, value):
         """ {'type': 'boolean'} """
         if readonly:
-            self._error(field, errors.READONLY_FIELD)
-            return True
+            if not self.is_normalized:
+                self._error(field, errors.READONLY_FIELD)
+                return True
+            # If the document was normalized (and therefore already been
+            # checked for readonly fields), we still have to return True
+            # if an error was filed if readonly is a priority rule.
+            if 'readonly' in self.priority_validations:
+                has_error = self.schema_error_tree.fetch_errors_from(
```

Comment:
Are you sure?  I thought the `SchemaErrorTree` allows to access errors for certain rules. In fact, I just tried that and indeed only get the error for `readonly`:

```python

d = errors.DocumentErrorTree()
s = errors.SchemaErrorTree()

e = errors.ValidationError(('a'), ('a', errors.READONLY_FIELD.rule),
    errors.READONLY_FIELD.code, errors.READONLY_FIELD.rule,
    None, None, None)
f = errors.ValidationError(('a'), ('a', errors.MIN_VALUE.rule),
    errors.MIN_VALUE.code, errors.MIN_VALUE.rule,
    None, None, None)
[tree.add(error) for tree in (d, s) for error in (e, f)]

# DocumentErrorTree returns all errors for the given field.
print(d.fetch_errors_from(('a',)))
>> [ValidationError @ 0x7f7686359ed0 ( document_path=a,schema_path=('a', 'min'),code=0x42,constraint=None,value=None,info=None ), ValidationError @ 0x7f76874f2150 ( document_path=a,schema_path=('a', 'readonly'),code=0x63,constraint=None,value=None,info=None )]

# SchemaErrorTree returns just the readonly error.
print(s.fetch_errors_from(('a', 'readonly'))))
>> [ValidationError @ 0x7f76874f2150 ( document_path=a,schema_path=('a', 'readonly'),code=0x63,constraint=None,value=None,info=None )]
```